documentation/source/factories.md

```diff
@@ -0,0 +1,207 @@
+from omegaconf import DictConfig
+import hydra
+
+from super_gradients import Trainer, init_trainer
+
+
+@hydra.main(config_path="recipes", version_base="1.2")
+def _main(cfg: DictConfig) -> None:
+    Trainer.train_from_config(cfg)
+
+
+def main() -> None:
+    init_trainer()  # `init_trainer` needs to be called before `@hydra.main`
+    _main()
+
+
+if __name__ == "__main__":
+    main()
+
+```
+
+## Under the Hood
+
+Until now, we saw how to use existing Factories, and how to register new ones.
+In some cases, you may want to create objects that would benefit from using the factories.
+
+### Basic
+The basic way to use factories as below.
+```
+from super_gradients.common.factories import TransformsFactory
+factory = TransformsFactory()
+my_transform = factory.get({'MyTransformName': {'prob':  0.7}})
+```
+You may recognize that the input passed to `factory.get` is actually the dictionary that we get after loading the recipe
+(See [Using Existing Factories](#using-existing-factories))
+
+### Recommended
+Factories become even more powerful when used with the `@resolve_param` decorator. 
+This feature allows functions to accept both instantiated objects and their dictionary representations. 
+It means you can pass either the actual python object or a dictionary that describes it straight from the recipe.
+
+```python
+class ImageNetDataset(torch_datasets.ImageFolder):
+    
+    @resolve_param("transforms", factory=TransformsFactory())
+    def __init__(self, root: str, transform: Transform):
+        ...
+```
+
+Now, `ImageNetDataset` can be passed both an instance of `MyTransform`
+
+```python
+my_transform = MyTransform(prob=0.7)
+ImageNetDataset(root=..., transform=my_transform)
+```
+
+And a dictionary representing the same object
+```python
+my_transform = {'MyTransformName': {'prob':  0.7}}
+ImageNetDataset(root=..., transform=my_transform)
+```
+
+This second way of instantiating the dataset combines perfectly with the concept `.yaml` recipes.
+
+## Supported Factory Types
+Until here, we focused on a single type of factory, `TransformsFactory`, 
+associated with the registration decorator `register_transform`. 
+
+SuperGradients supports a wide range of factories, used throughout the training process, 
+each with its own registering decorator.
+ 
+SuperGradients offers various types of factories, and each is associated with a specific registration decorator.
+
+``` python
+from super_gradients.common.factories import (
+    register_model
+    register_kd_model
+    register_detection_module
+    register_metric
+    register_loss
+    register_dataloader
+    register_callback
+    register_transform
+    register_dataset
+    register_pre_launch_callback
+    register_unet_backbone_stage
+    register_unet_up_block
+    register_target_generator
+    register_lr_scheduler
+    register_lr_warmup
+    register_sg_logger
+    register_collate_function
+    register_sampler
+    register_optimizer
+    register_processing
```

Comment:
add comma ","

Reply:
Fixed in [6b8e8ee](https://github.com/Deci-AI/super-gradients/pull/1395/commits/6b8e8ee8aee63068e1aee19cef7b314c06a44373)